lib/bolt/pal.rb

```diff
@@ -143,6 +143,35 @@ def list_tasks
       end
     end
 
+    def parse_param(val, type, compiler)
+      # TODO: is there a constant for this in puppet?
+      any_string = compiler.evaluate_string('String[0]')
+      unless any_string.assignable?(type)
```

Comment:
This is backward, should be `type.assignable?(any_string)` (ie. does the type accept an instance of string?)

Reply:
That won't work: String[1] will not accept all instances of String[0] an so is not assignable. I don't think there is actually an assignability test we can use that would work for something like:
Optional[Pattern[/<uuid>/]]